sdk/src/client/api/block_builder/input_selection/requirement/amount.rs

```diff
@@ -72,379 +117,110 @@ impl InputSelection {
-        if amount_selection.fulfil(self, base_inputs)? {
-            return Ok(true);
-        }
-
-        Ok(false)
+        Ok(input_amount >= *output_amount)
     }
 
-    fn reduce_funds_of_chains(&mut self, amount_selection: &mut AmountSelection) -> Result<(), Error> {
-        // Only consider automatically transitioned outputs.
-        for output in self.added_outputs.iter_mut() {
-            let diff = amount_selection.missing_amount();
-            let amount = output.amount();
-            let minimum_amount = output.minimum_amount(self.protocol_parameters.storage_score_parameters());
-
-            let new_amount = if amount >= diff + minimum_amount {
-                amount - diff
+    fn sort_available_inputs_for_amount(&mut self, missing_amount: u64) {
+        // Establish the order in which we want to pick an input
+        let sort_order_type = [
+            BasicOutput::KIND,
+            AccountOutput::KIND,
+            NftOutput::KIND,
+            FoundryOutput::KIND,
+            AnchorOutput::KIND,
+            DelegationOutput::KIND,
+        ]
+        .into_iter()
+        .zip(0..)
+        .collect::<HashMap<_, _>>();
+        let amount_sort = |output: &Output| {
+            let mut amount = output.amount();
+            if let Some(sdruc) = sdruc_not_expired(output, self.latest_slot_commitment_id.slot_index()) {
+                amount -= sdruc.amount();
+            }
+            // If the amount is greater than the missing amount, we want the smallest ones first
+            if amount >= missing_amount {
+                (false, amount)
+            // Otherwise, we want the biggest first
             } else {
-                minimum_amount
-            };
-
-            // TODO check that new_amount is enough for the storage cost
-
-            // PANIC: unwrap is fine as non-chain outputs have been filtered out already.
-            log::debug!(
-                "Reducing amount of {} to {} to fulfill amount requirement",
-                output.chain_id().unwrap(),
-                new_amount
-            );
-
-            let new_output = match output {
-                Output::Account(output) => AccountOutputBuilder::from(&*output)
-                    .with_amount(new_amount)
-                    .finish_output()?,
-                Output::Foundry(output) => FoundryOutputBuilder::from(&*output)
-                    .with_amount(new_amount)
-                    .finish_output()?,
-                Output::Nft(output) => NftOutputBuilder::from(&*output)
-                    .with_amount(new_amount)
-                    .finish_output()?,
-                _ => panic!("only account, nft and foundry can be automatically created"),
-            };
-
-            amount_selection.outputs_sum -= amount - new_amount;
-            *output = new_output;
-
-            if amount_selection.missing_amount() == 0 {
-                return Ok(());
+                (true, u64::MAX - amount)
```

Comment:
Why is this subtracted and not just used directly?

Reply:
We want to sort it backwards, and we can't use -amount because it's an unsigned value

Reply:
But why don't keep it as the positive amount it is and change it in the sort function?

Reply:
I don't understand?

Reply:
You can order them by amount in both ways without subtracting from u64?

Reply:
Maybe I'm missing something, but you could just do .reverse() or change the order in the compare?
```
fn main() {
    let mut numbers = vec![3, 4, 2, 1, 5];
    numbers.sort();
    println!("{numbers:?}");
    numbers.reverse();
    println!("{numbers:?}");
    numbers.sort_by(|a, b| b.cmp(a));
    println!("{numbers:?}");
}

```

Reply:
No, because half of the amounts are sorted high-to-low and half are sorted low-to-high

Reply:
Why is this done?

Reply:
Because if we have amounts higher than the needed amount, we want to get the smallest one. But if we have amounts lower then we want to get the highest. Both avoid doing excess work.

Reply:
> Because if we have amounts higher than the needed amount, we want to get the smallest one. 

So let's say we have outputs with amounts: 1, 2, 4, 7
Provided amount is 6 would it select the inputs with 1, 2, 4?
And if provided amount is 8 it would select 7, 1?

Reply:
Thanks for explaining